ajax_with_jquery_exercise/nacker_hews_AJAX/app.js

```diff
@@ -0,0 +1,292 @@
+      hackerNews.forEach(function(value) {
+        $.get("https://hacker-news.firebaseio.com/v0/item/" + value + ".json?print=pretty")
+        .then(function(res){
+          console.log(res);
+          let $li = $("<li>");
+          let $star = $ ("<span>", {
+            class: "glyphicon glyphicon-star-empty hide-all",
+          })
+          $star.data("story_id",res.id);
+          let $text = $("<a>", {
+            class: "topic-text",
+            target: "blank",
+            href: res.url,
+            text: res.title
+          });
+          let $link = $("<span>", {
+            class: "class-filter",
+            text: `(${res.url})`
+          });
+          $li.append($star,$text,$link);
+          $topiclist.append($li);
+          hackerNews = [];
+          //Add star if auto logged in
+          if (localStorage.getItem("Nacker")) {
+            $(".glyphicon").removeClass("hide-all");    
+          }
+        })
+      })
+    })
+  } 
+  //Function to fetch new 25 results for fetch news.
+  function fetchNewNews() {
+    $topiclist.empty();
+    $.get("https://hacker-news.firebaseio.com/v0/newstories.json?print=pretty")
+    .then(function(res){
+      hackerNews = hackerNews.concat(res.slice(0,25));
+      hackerNews.forEach(function(value) {
+        $.get("https://hacker-news.firebaseio.com/v0/item/" + value + ".json?print=pretty")
+        .then(function(res){
+          console.log(res);
+          let $li = $("<li>");
+          let $star = $ ("<span>", {
+            class: "glyphicon glyphicon-star-empty hide-all",
+          })
+          $star.data("story_id",res.id);
+          let $text = $("<a>", {
+            class: "topic-text",
+            target: "blank",
+            href: res.url,
+            text: res.title
+          });
+          let $link = $("<span>", {
+            class: "class-filter",
+            text: `(${res.url})`
+          });
+          $li.append($star,$text,$link);
+          $topiclist.append($li);
+          hackerNews = [];
+          //Add star if auto logged in
+          if (localStorage.getItem("Nacker")) {
+            $(".glyphicon").removeClass("hide-all");    
+          }
+        })
+      })
+    })
+  } 
+  //Function to fetch favorites
+
+  function fetchFavorites() {
+    $topiclist.empty();
+    $.ajax({
+          type: "Get",
+          url: "https://hn-favorites.herokuapp.com/stories.json",
+          headers: {
+            authorization: JSON.parse(localStorage.getItem("Nacker"))[0]
+          }
+        }).then(function(res) {
+          console.log(res);
+          hackerNews = hackerNews.concat(res);
+          hackerNews.forEach(function(value) {
+          let $li = $("<li>");
+          let $star = $ ("<span>", {
+            class: "glyphicon glyphicon-star",
+          });
+          $star.data("story_id",value.story_id);
+          $star.data("favorite_id",value.id);
+          let $text = $("<a>", {
```

Comment:
this also looks pretty similar to what you're doing in the above three functions. Maybe you could write a helper function to add a list item?